lightning-invoice/src/payment.rs

```diff
@@ -367,13 +439,13 @@ where
 				PaymentSendFailure::PathParameterError(_) => Err(e),
 				PaymentSendFailure::AllFailedRetrySafe(_) => {
 					let mut payment_cache = self.payment_cache.lock().unwrap();
-					let retry_count = payment_cache.get_mut(&payment_hash).unwrap();
-					if *retry_count >= self.retry_attempts.0 {
-						Err(e)
+					let payment_attempts = payment_cache.get_mut(&payment_hash).unwrap();
+					if self.retry.is_retryable_now(payment_attempts) {
+						payment_attempts.counts += 1;
+						Ok(self.pay_internal(params, payment_hash, send_payment)?)
```

Comment:
Don't we need to drop the lock in the retry branch? Looks like you swapped the branch order but left the lock-drop in the other branch now.

Reply:
Sorry totally my mistake